src/shared/firewall-util-nft.c

```diff
@@ -812,6 +836,70 @@ static int fw_nftables_recreate_table(sd_netlink *nfnl, int af, sd_netlink_messa
         return 0;
 }
 
+static int nft_message_add_setelem_ip6range(sd_netlink_message *m,
+                                            const union in_addr_union *source,
+                                            unsigned int prefixlen) {
+        struct in6_addr start, end;
+        uint8_t carry = 0;
+        int r, i, j;
+
+        assert(prefixlen <= 128);
+
+        for (i = 0, j = 15; i < 16; i++) {
+                uint8_t nm;
+
+                nm = 0xFF;
+                if (prefixlen < 8)
+                        nm = 0xFF << (8 - prefixlen);
+
+                start.s6_addr[i] = source->in6.s6_addr[i] & nm;
+                end.s6_addr[i] = start.s6_addr[i];
+                if (prefixlen <= 8 && j == 15) {
+                         carry = 1u << (8 - prefixlen);
+                         j = i;
+                }
+
+                if (prefixlen >= 8)
+                         prefixlen -= 8;
+                else
+                         prefixlen = 0;
+        }
+
+        r = sd_nfnl_nft_message_add_setelem(m, 0, &start, sizeof(start), NULL, 0);
+        if (r < 0)
+                return r;
+
+        r = sd_nfnl_nft_message_add_setelem_end(m);
+        if (r < 0)
+                return r;
+
+        for (; j >= 0; j--) {
+                uint16_t overflow = end.s6_addr[j] + carry;
+
+                end.s6_addr[j] = overflow;
+                if (overflow <= 0xff)
+                        break;
+                carry = 1;
+        }
+
+        if (memcmp(&start, &end, sizeof(start)) > 0)
+                memset(&end, 0, sizeof(end));
```

Comment:
we have a macro "zero()" that should be used here. Main reason for it is that it avoids the typical confusion about what is second and what third argument

Reply:
done.